src/module.c

```diff
@@ -13512,6 +13512,35 @@ const char* RM_GetInternalSecret(RedisModuleCtx *ctx, size_t *len) {
     return secret;
 }
 
+/* CONFIG SET
+ * Set the value of the config. If the config does not exist or is a module
+ * config return REDISMODULE_ERR */
+int RM_ServerConfigSet(RedisModuleCtx *ctx, const char *name, const char *value) {
```

Comment:
do we wanna consider adding specific APIs for different primitive types (bool, long)?
in many cases we can avoid an excessive string printing and parsing..

do we wanna consider an API to let the module iterate over the configs? or maybe in the future.
with `RM_Call("CONFIG", "GET", "*")` it's possible, but the point here is that we want to avoid the need for RM_Call

Reply:
If needed by redis modules - I'll add. IIUC this is scarcely used, i.e performance and ergonomics should not be a problem for now so I vote it to stay this way as API is as simple as possible.

Reply:
CONFIG GET might be used a lot, possibly every time the module needs the value.. unless the module implements some caching.
i'm not sure how we can conclude how's is used (where did you look fort that?), and even if we could, the way it's used now, doesn't reflect what we'll have in the future.
unless we expose limiting API and modules have to work around our limitations, which is why i always prefer the API to be few steps ahead.
and what's worse, the API is here to stay (we can't break it)
i suggest to learn from the APIs we added around INFO and see if we can do something similar.